### PR TITLE
Added CCU2 HomematicIP support

### PR DESCRIPTION
**Pull request by gerrieg**
HomematicIP is disabled by default, to enable it:

```
homematic:homematicIP.enabled=true
```

### CODE REVIEW

**Comment by gerrieg** [comment]
@teichsta 
Any chance to get this merged?

**Comment by kaikreuzer** [comment]
@gerrieg Please note that @teichsta [is no longer a maintainer of openHAB](https://community.openhab.org/t/growing-the-number-of-active-maintainers/4278/47?u=kai), so you should not wait for him.
Maybe some of the other @openhab/openhab-1-add-ons-maintainers would be willing to review and merge this PR?
In any case, it seems the PR needs to be rebased, before it can be merged - so you might want to do so.

File: bundles/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/common/HomematicConfig.java
Comment (9037568) [inline bundles/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/common/HomematicConfig.java]: I think this should say "default is false"

File: bundles/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/common/HomematicConfig.java
Comment (9037568) [inline bundles/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/common/HomematicConfig.java:191]: This is removing a pre-existing public method. Should probably leave this in and mark it as deprecated.

File: distribution/openhabhome/configurations/openhab_default.cfg
Comment (9037568) [inline distribution/openhabhome/configurations/openhab_default.cfg]: "default is false"

File: bundles/binding/org.openhab.binding.homematic/.classpath
Comment (teichsta) [inline bundles/binding/org.openhab.binding.homematic/.classpath]: remove sourcepath

**Comment by teichsta** [comment]
thanks for clarifying @kaikreuzer … nevertheless i am going to review your changes until weekend.

A general question: is this "just" a backport of the OH2 implementation @gerrieg ? This would reduce the review efforts.

Thanks, Thomas E.-E.

**Comment by gerrieg** [comment]
This is not a backport, i implemented HomematicIP in OH1 first and then i ported it to OH2. But many things are the same.

**Comment by gerrieg** [comment]
rebase done, now it should be mergeable

**Comment by marfnk** [comment]
Nice work!
Works with HMIP-WTH Thermostat and a CCU2 like a charm.

**Comment by danielkr123** [comment]
Does anyone have an example of an OpenHab item configuration with HomematicIP? I have a HMIP-SWSD smoke alarm but simply adding it as if I had a Homematic device does not work:
`String  SZ_Rauchmelder  " [%b]" <fire>  { binding="address=000A55699D1234, channel=1, parameter=STATE" }`
What makes me doubt that it can work like that is that the serial numbers look quite different to the Homematic devices.

**Review by 9037568 — CHANGES_REQUESTED** [review]
A couple tiny tweaks. Also please run all source through the eclipse formatter.

**Pull request closed without merging**
